# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1322344739 0
#      Sat Nov 26 21:58:59 2011 +0000
# Node ID fc6ebd31f40f17b26a2430b0718f8d76b1f2fa13
# Parent  3cb8f1fe108c54a18da269496a6a4e21319b1376
Support linestyleorder as cellstr.

* scripts/plot/private/__next_line_style__.m: Support linestyleorder value as
cell array of strings.

diff --git a/scripts/plot/private/__next_line_style__.m b/scripts/plot/private/__next_line_style__.m
--- a/scripts/plot/private/__next_line_style__.m
+++ b/scripts/plot/private/__next_line_style__.m
@@ -34,17 +34,20 @@ function [linestyle, marker] = __next_li
     if (nargin == 1)
       if (ischar (reset) && strncmp (reset, "incr", 4))
         if (isempty (style_rotation))
           error ("__next_line_style__: style_rotation not initialized");
         elseif (++style_index > num_styles)
           style_index = 1;
         endif
       elseif (reset || isempty (style_rotation))
-        style_rotation = strsplit (get (gca (), "linestyleorder"), "|");
+        style_rotation = get (gca (), "linestyleorder");
+        if (ischar (style_rotation))
+          style_rotation = strsplit (style_rotation, "|");
+        endif
         num_styles = length (style_rotation);
         style_index = 1;
       endif
     elseif (! isempty (style_rotation))
       options = __pltopt__ ("__next_line_style__",
                             style_rotation (style_index));
       linestyle = options.linestyle;
       marker = options.marker;
